warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/reindex/ShardReindexJob.java

```diff
@@ -200,7 +273,47 @@ private Job setupJob() throws IOException, ParseException, AccumuloException, Ta
         return j;
     }
 
-    public static Collection<Range> buildRanges(String start, String end, int splitsPerDay) throws ParseException {
+    private Set<String> getSplitsFromMetadata(Properties accumuloProperties, String tableName, Range r) {
+        Set<String> rangeSplits = new HashSet<>();
+
+        AccumuloClient client = Accumulo.newClient().from(accumuloProperties).build();
+        String tableId = client.tableOperations().tableIdMap().get(tableName);
+
+        if (tableId == null) {
+            throw new RuntimeException("Could not locate table: '" + tableName + "'");
+        }
+
+        try {
+            Scanner s = client.createScanner("accumulo.metadata");
```

Comment:
Ideally, we'd close the client and the scanner if an exception occurs to, with a finally block or better yet with a try-with-resources:

```
try (AccumuloClient client = .... ;
      Scanner metadataScanner = .... ) {
        // rest of your stuff
}
```